src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -97,38 +93,61 @@ public void initializeConfigSupplier() {
       );
   }
 
-  public String getLatestConfiguration() {
-    if (configurationToken == null) {
-      return defaultConfigValue;
-    }
+  public JSONObjectAdapter getLatestConfiguration() {
     try {
+      if (configurationToken == null) {
+        logger.log(Level.SEVERE, "returning default config");
+        return new JSONObjectAdapterImpl(DEFAULT_CONFIG_VALUE);
+      }
       GetLatestConfigurationRequest latestConfigRequest =
         new GetLatestConfigurationRequest()
           .withConfigurationToken(configurationToken);
       GetLatestConfigurationResult latestConfigResponse =
         appConfigDataClient.getLatestConfiguration(latestConfigRequest);
       configurationToken = latestConfigResponse.getNextPollConfigurationToken();
-      ByteBuffer configData = latestConfigResponse.getConfiguration();
-      String newConfigValue = new String(
-        configData.array(),
+      ByteBuffer readOnlyConfigData = latestConfigResponse
+        .getConfiguration()
+        .asReadOnlyBuffer();
+      byte[] bytes = new byte[readOnlyConfigData.remaining()];
+      readOnlyConfigData.get(bytes);
+      String newConfigString = new String(
+        bytes,
```

Comment:
You can avoid instantiating and filling an array if you use a CharBuffer, though it's probably doing the exact same thing that you're doing here under-the-hood. I would just prefer to let the Java IO abstractions handle that

https://stackoverflow.com/a/23552156

https://www.mindprod.com/jgloss/bytebuffer.html

```suggestion
       CharBuffer charBuffer = StandardCharsets.UTF_8.decode(readOnlyConfigData);
       String newConfigString = charBuffer.toString();
```